### ISSUE TITLE
Multi-Thread game validation scene

### ISSUE THREAD

**Issue by bahusafoo** [bahusafoo/Atavism-Game-Launcher-for-Windows#2]
Implement file validation as a background job/on a separate thread + show a progress bar or at least a stock spinner while that task is running.  Currently the validation happens on the main thread, freezing the compiled game executable (Not Responding) until all files have been worked through - this is not ideal and is even more noticeable in larger sized games.

**Comment by bahusafoo** [comment]
Long loading times in valdiation scene occur due to the validation scene being single threaded and working on a large list of large files.  The validation scene is still utilizing the first iteration ever created and hasn't had any enhancement focus given at all. 
 Presently, The bigger your game client, the worse this will be.  Validation is important, as it ensures your game clients don't have anything modded or extra added to them other than what you deliver.  There is a feature enhancement planned already for this that will make this way faster and display a status as validation occurs (currently, the process appears to become unresponsive as file validation happens on the main thread, multithreading will allow us to display a status indicator, at least a spinner)

**Comment by bahusafoo** [comment]
Multi-Threading has been added to the game validation scene.  This lays the framework for chunking up validation work into smaller jobs which can run in parallel, which will decrease the time for validation to occur while launching games (future work).  Opening a separate issue for validation job chunking, closing this one out as the "not responding" issue itself has been resovled.
